# RnaSeqMetaAnalyst
![image](https://github.com/user-attachments/assets/3f5cde0a-61b9-4ed2-af77-9ffa789507de)

A bioinformatics pipeline to perfrom Meta-analysis of RNA-Seq data using uniform processing.
This repository contains a complete, reproducible RNA-Seq analysis pipeline packaged in a Docker container. It automates alignment (HISAT2, STAR, Bowtie2), Quantificatin (featureCount, htseq-count, rsem) and differential gene expression analysis using DESeq2 and edgeR.

ğŸ“¦ Features
ğŸ” End-to-end automation: fromAlignment to DEG analysis

ğŸ³ Dockerized: no dependency issues

ğŸ§¬ Supports HISAT2, STAR, Bowtie2 for alignment

ğŸ“Š Generates Excel summaries for DEG results (edgeR & DESeq2)

ğŸ“ Output organized in user-friendly structure


# Requirements
## Files

## Tools 
## ğŸ“ Project Directory Structure

```
project_root/
â”‚
â”œâ”€â”€ reference/                     # Reference genome and annotation
â”‚   â”œâ”€â”€ genome.fa                  # Genome FASTA file
â”‚   â”œâ”€â”€ genome.gtf
|   |               # Annotation GTF file
â”‚   â”œâ”€â”€ hisat_index/               # HISAT2 index
â”‚   â”œâ”€â”€ bowtie2_index/             # Bowtie2 index
â”‚   â””â”€â”€ star_index/                # STAR index
â”‚
â”œâ”€â”€ data/                          # Raw sequencing files
â”‚   â”œâ”€â”€ sample1_R1.fastq.gz
â”‚   â”œâ”€â”€ sample1_R2.fastq.gz
â”‚   â””â”€â”€ ...
â”‚
â”œâ”€â”€ sampleinfo/
â”‚   â””â”€â”€ sampleinfo.txt             # Sample condition metadata
â”‚
â”œâ”€â”€ DEG/                           # DEG results (edgeR/DESeq2)
â”‚   â””â”€â”€ (generated output folders per comparison)
â”‚
â”œâ”€â”€ bioinfo_pipeline/
â”‚   â”œâ”€â”€ Dockerfile                 # Docker build file
â”‚   â””â”€â”€ scripts/                   # Pipeline scripts
â”‚       â”œâ”€â”€ master.py              # Master controller script
â”‚       â”œâ”€â”€ run_all_edgeR.sh       # edgeR batch runner
â”‚       â”œâ”€â”€ run_all_deseq2.sh      # DESeq2 batch runner
â”‚       â”œâ”€â”€ run_edgeR_analysis.R   # edgeR DEG logic
â”‚       â”œâ”€â”€ run_deseq_222.R        # DESeq2 DEG logic
â”‚       â”œâ”€â”€ hisat_ok.sh            # HISAT2 aligner
â”‚       â”œâ”€â”€ bowtie.sh              # Bowtie2 aligner
â”‚       â”œâ”€â”€ star.sh                # STAR aligner
â”‚       â””â”€â”€ (any other helper scripts)
```
<pre> 
   # Clone the repository
git clone https://github.com/bioinfokushwaha/RnaSeqMetaAnalyst.git

# Move into the pipeline directory
cd RnaSeqMetaAnalyst/bioinfo_pipeline

# Build the Docker image
docker build -t rnaseq-metaanalyst .

# Run the pipeline (adjust paths if needed)
docker run -it -v "$PWD/data":/data -v "$PWD/counts":/counts  -v "$PWD/output":/output  rnaseq-metaanalyst python /opt/bioinfo/scripts/master.py
 </pre>
